### PR TITLE
Conform to PEP8 standards

### PR DESCRIPTION
**Pull request by teaearlgraycold**
## What does this PR do?

* Provides Python code that (for the most part) conforms to PEP8 standards.
  - Line length wasn't taken too seriously, although I still tried to stick with 80 characters when reasonable
  - The code copied from DEAP wasn't treated as strictly as the rest of the code
* Improves code readability through functional decomposition

### CODE REVIEW

**Comment by rhiever** [comment]
This will take a while to review, but let's try to get this merged ASAP so it doesn't slow down other development efforts. @weixuanfu2016, can you please take a look in the next day or two to see if you find any issues? I will plan to do so as well.

**Comment by weixuanfu2016** [comment]
OK, will do

File: tests.py [hunk 1]
```diff
@@ -228,87 +237,82 @@ def test_random_ind_2():
...
-        pass
+    assert_raises(RuntimeError, tpot_obj.score, testing_features, testing_classes)
 
 
 def test_score_2():
-    """Assert that the TPOTClassifier score function outputs a known score for a fix pipeline"""
-
+    """Assert that the TPOTClassifier score function outputs a known score for a fix pipeline."""
     tpot_obj = TPOTClassifier()
     known_score = 0.977777777778  # Assumes use of the TPOT balanced_accuracy function
```
Comment (weixuanfu2016) [inline tests.py]: May need to delete or update this comment since the default scoring function is `accuracy` instead of `balanced_accuracy`

File: tpot/driver.py [hunk 2]
```diff
@@ -65,176 +65,406 @@ def float_range(value):
...
+    return input_data
 
-    training_features, testing_features, training_classes, testing_classes = \
-        train_test_split(features, input_data[args.TARGET_NAME], random_state=args.RANDOM_STATE)
 
-    if args.TPOT_MODE == 'classification':
-        tpot_type = TPOTClassifier
-    else:
-        tpot_type = TPOTRegressor
+def _impute_missing_values(features, missing_value):
```
Comment (weixuanfu2016) [inline tpot/driver.py]: I suggest that this imputation function could also be added into interactive mode by moving it to base.py

File: tpot/driver.py [hunk 3]
```diff
@@ -65,176 +65,406 @@ def float_range(value):
...
+        disable_update_check=args.DISABLE_UPDATE_CHECK
+    )
 
     print('')
 
     tpot.fit(training_features, training_classes)
 
-    if args.VERBOSITY in [1, 2] and tpot._optimized_pipeline:
-        training_score = max([tpot._pareto_front.keys[x].wvalues[1] for x in range(len(tpot._pareto_front.keys))])
+    if args.VERBOSITY < 3 and tpot._optimized_pipeline:
```
Comment (weixuanfu2016) [inline tpot/driver.py]: Here need to be [1, 2] for no stdout when `verbosity=0`

File: tpot/gp_deap.py [hunk 4]
```diff
@@ -71,35 +71,36 @@ def varOr(population, toolbox, lambda_, cxpb, mutpb):
     for _ in range(lambda_):
         op_choice = np.random.random()
         if op_choice < cxpb:            # Apply crossover
-            idxs = np.random.randint(0, len(population),size=2)
+            idxs = np.random.randint(0, len(population), size=2)
             ind1, ind2 = toolbox.clone(population[idxs[0]]), toolbox.clone(population[idxs[1]])
             ind_str = str(ind1)
             num_loop = 0
-            while ind_str == str(ind1) and num_loop < 50 : # 50 loops at most to generate a different individual by crossover
+            while ind_str == str(ind1) and num_loop < 50:  # 50 loops at most to generate a different individual by crossover
```
Comment (weixuanfu2016) [inline tpot/gp_deap.py]: May also used `NUM_TESTS` instead as decorators.py

**Comment by teaearlgraycold** [comment]
Oops. the imputation function was mistakenly left in. I'll remove that and
update the PR.

File: tpot/gp_deap.py (hunk 4 above)
Comment (teaearlgraycold) [inline tpot/gp_deap.py]: Sure, might as well get rid of the magic numbers from DEAP.

File: tests.py [hunk 5]
```diff
@@ -59,28 +57,46 @@
...
-TPOTSelectKBest,TPOTSelectKBest_args = TPOTOperatorClassFactory(test_operator_key,
-                                            classifier_config_dict[test_operator_key])
+TPOTSelectKBest, TPOTSelectKBest_args = TPOTOperatorClassFactory(
+    test_operator_key,
+    classifier_config_dict[test_operator_key]
+)
+
+
+# http://stackoverflow.com/questions/5595425/
+def is_close(a, b, rel_tol=1e-09, abs_tol=0.0):
```
Comment (rhiever) [inline tests.py]: We should use numpy.allclose here: https://docs.scipy.org/doc/numpy/reference/generated/numpy.allclose.html

File: tpot/base.py [hunk 6]
```diff
@@ -563,12 +581,17 @@ def score(self, testing_features, testing_classes):
...
-        return abs(SCORERS[self.scoring_function](self._fitted_pipeline,
-            testing_features.astype(np.float64), testing_classes.astype(np.float64)))
+        # If the scoring function is a string, we must adjust to use the sklearn
+        # scoring interface
+        score = SCORERS[self.scoring_function](
+            self._fitted_pipeline,
+            testing_features.astype(np.float64),
+            testing_classes.astype(np.float64)
+        )
+        return abs(score)
```
Comment (rhiever) [inline tpot/base.py]: Is returning `abs` of the score the correct thing to do here? The blame graph shows that we added it when we added regressors, but what is the thinking behind it?

File: tpot/built_in_operators.py [hunk 7]
```diff
@@ -57,19 +52,27 @@ def transform(self, X, y=None):
...
         return X_transformed
 
+
 class CombineDFs(object):
-    """Operator to combine two DataFrames"""
+    """Combine two DataFrames."""
 
     @property
     def __name__(self):
+        """Instance ame is the same as the class name."""
```
Comment (rhiever) [inline tpot/built_in_operators.py]: Typo here: `ame`

File: tpot/config_classifier.py [hunk 8]
```diff
@@ -168,22 +160,22 @@
...
         'score_func': {
             'sklearn.feature_selection.f_classif': None
-            } # read from dependencies ! need add an exception in preprocess_args
+        }  # read from dependencies ! need add an exception in preprocess_args
 
     },
 
     'sklearn.feature_selection.SelectKBest': {
-        'k': range(1, 100), # need check range!
+        'k': range(1, 100),  # TODO: Check range
```
Comment (rhiever) [inline tpot/config_classifier.py]: @weixuanfu2016, please file an issue for this TODO.

Comment (weixuanfu2016) [inline tpot/config_classifier.py]: OK

**Comment by rhiever** [comment]
OK. After the two issues I mentioned above are settled, I'm 👍 on merging.

File: tpot/base.py (hunk 6 above)
Comment (teaearlgraycold) [inline tpot/base.py]: IIRC some scorers procude scores in the range `[0, ∞)`, but the regressors can have scores in the range `(-∞, ∞)`.

Comment (rhiever) [inline tpot/base.py]: OK. I'll file a separate issue for this topic since it's not particularly related to this PR.

**Comment by teaearlgraycold** [comment]
Hold on, ignore that commit. I'm going to remove something and rebase.

File: tpot/base.py [hunk 9]
```diff
@@ -293,9 +294,8 @@ def _setup_config(self, config_dict):
                     )
             else:
                 try:
-                    with open(config_dict, 'r') as input_file:
-                        file_string = input_file.read()
-                    self.config_dict = eval(file_string[file_string.find('{'):(file_string.rfind('}') + 1)])
+                    config_module = import_module(config_dict)
+                    self.config_dict = config_module.tpot_config
```
Comment (rhiever) [inline tpot/base.py]: Was this change intended? I thought this was a separate PR.

Comment (teaearlgraycold) [inline tpot/base.py]: See my comment above

Comment (rhiever) [inline tpot/base.py]: Hopefully that doesn't change too much. :-)

**Comment by rhiever** [comment]
PR looks good to merge. Any outstanding issue before I merge it?

**Comment by teaearlgraycold** [comment]
Don't think so. Merge away
